tools/build_framework.sh

```diff
@@ -10,7 +10,8 @@ fi
 
 FRAMEWORK_NAME=$1
 FRAMEWORK_DIR=$2
```

Comment:
Actually I tried switching cassandra to use this just a bit ago and found that it's pretty close as-is. Disregard earlier comment as it wouldn't really make things easier or more flexible. Instead, how about doing the following:
- Make publish step required, maybe even have it be the first parameter. Have a value of `none` mean to skip the publish step.
- Change `BUILD_DIR` to `ARTIFACT_FILES`, and have it be a variable length argument at the end.

With these changes this section could look like this (with weird handling to make the varargs work):

```
PUBLISH_STEP=$1
shift
FRAMEWORK_NAME=$1
shift
FRAMEWORK_DIR=$1
shift
ARTIFACT_FILES=$@
```

With this, `${ARTIFACT_FILES}` could be passed directly to the publish step at the end, allowing the caller to pass whatever artifact files they need to publish in the package.

Usage would look like:
```
build_framework.sh \
  none \
  kafka \
  frameworks/kafka/ \
  frameworks/kafka/scheduler/scheduler.zip \
  frameworks/kafka/executor/executor.zip
  [...]
```